app/controllers/galleries_controller.rb

```diff
@@ -48,6 +54,16 @@ def search_api_for_image_metadata(images)
     search_results(blacklight_api_params_for_images(images)).last
   end
 
+  def gallery_hero_image_from_cache
+    Rails.cache.fetch(cache_key(@body_cache_key) + '/hero_image_url')
+  end
+
+  def set_gallery_hero_image_cache(image)
+    image_document = @documents.detect { |document| document.fetch(:id, nil) == image.europeana_record_id }
+    @hero_image_url = image_document['edmIsShownBy'].first
+    Rails.cache.write(cache_key(@body_cache_key) + '/hero_image_url', @hero_image_url, expires_in: 24.hours)
```

Comment:
@lbiedinger should we cache this for 25 hours? The cached HTML would always then expire first and the cached hero image URL get overwritten when the former is next generated.